File: BOJ/1940.cpp
#include <stdio.h>
#include <algorithm>
using namespace std;
int make[15001];
int m,n,cnt;
int main() {
	scanf("%d", &n);
	scanf("%d", &m);
	for (int i = 1; i <= n; i++) { scanf("%d", &make[i]);}
	sort(make + 1, make + n + 1);
	int left = 1;
	int right = n;
	for (; left < right;) {
		if (make[left] + make[right] < m) {
			left++;
			continue;
		}
		else if (make[left] + make[right] > m) {
			right--;
			continue;
		}
		else if (make[left] + make[right] == m) {
			cnt++;
			right--;
		}
	}
	printf("%d\n", cnt);
	return 0;
}

File: BOJ(Java 공부용)/1940.java
import java.io.*;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    static int[] gox = {0, 1, -1, 0, 1, 1, -1, -1};
    static int[] goy = {1, 0, 0, -1, 1, -1, 1, -1};

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());
        int m = Integer.parseInt(br.readLine());
        int[] arr = new int[n];
        StringTokenizer st = new StringTokenizer(br.readLine());
        for (int i = 0; i < n; i++) arr[i] = Integer.parseInt(st.nextToken());
        Arrays.sort(arr);
        int left = 0, right = n - 1, cnt = 0;
        while (left < right) {
            if (arr[left] + arr[right] == m) {
                cnt++;
                right--;
            } else if (arr[left] + arr[right] < m) {
                left++;
            } else if (arr[left] + arr[right] > m) {
                right--;
            }
        }
        bw.write(cnt + "\n");
        bw.flush();
        bw.close();
    }
}